packages/astro/src/core/build/index.ts

```diff
@@ -72,10 +73,13 @@ class AstroBuilder {
     );
     this.viteConfig = viteConfig;
     const viteServer = await vite.createServer(viteConfig);
+    // Listen to the server so that we can make fetch requests if needed.
+    await viteServer.listen(this.port);
     this.viteServer = viteServer;
     debug(logging, 'build', timerMessage('Vite started', timer.viteStart));
 
     timer.loadStart = performance.now();
+    const fetch = createFetch(this.origin);
```

Comment:
This is a localized fetch so you can do `fetch('/path/to/file.css')`, and can be passed to plugins that need it.